src/OSPSuite.Presentation/Mappers/DiffItemToDiffItemDTOMapper.cs

```diff
@@ -86,7 +89,24 @@ private string presentWithDetails(string presentDetails)
 
       public void Visit(PropertyValueDiffItem propertyDiffItem)
       {
-         updateDiffItem(propertyDiffItem.FormattedValue1, propertyDiffItem.FormattedValue2, propertyDiffItem.PropertyName, objectNameFrom(propertyDiffItem), propertyDiffItem.CommonAncestor, itemIsMissing: false);
+         Func<PathElements> pathElementsMapper = () => mapPathElementsAsIEntity(propertyDiffItem.CommonAncestor);
+
+         if (propertyDiffItem.Object1.GetType() == propertyDiffItem.Object2.GetType() && propertyDiffItem.Object1.IsAnImplementationOf<PathAndValueEntity>())
+            pathElementsMapper = () => mapPathElementsAsPathAndValueEntity(propertyDiffItem);
+
+         updateDiffItem(propertyDiffItem.FormattedValue1, propertyDiffItem.FormattedValue2, propertyDiffItem.PropertyName, objectNameFrom(propertyDiffItem), itemIsMissing: false, pathElementsMapper);
+      }
+
+      private PathElements mapPathElementsAsPathAndValueEntity(PropertyValueDiffItem propertyDiffItem)
+      {
+         var object1 = propertyDiffItem.Object1 as PathAndValueEntity;
```

Comment:
downcast here. This methods required those objects to be PathAndValueEntity so this should never be the case